dashboard/app/assets/javascripts/header.js

```diff
@@ -73,6 +73,20 @@ dashboard.buildHeader = function (stageData, progressData, currentLevelId, userI
     }
     progressContainer.append(div).append('\n');
   });
+
+  $.ajax('/api/user_progress/' + scriptName + '/' + stageIndex).done(function (data) {
+    // Merge progress from server (loaded via AJAX)
+    var serverProgress = data || {};
+    Object.keys(serverProgress).forEach(function (levelId) {
+      if (serverProgress[levelId] !== clientProgress[levelId]) {
+        var status = mergedActivityCssClass(clientProgress[levelId], serverProgress[levelId]);
+
+        // Clear the existing class and replace
+        $('#header-level-' + levelId).attr('class', 'level_link ' + status);
+      }
+    });
+  });
+
```

Comment:
There is some duplication here relative to https://github.com/code-dot-org/code-dot-org/blob/0f7341a166049c349dc7227a13dc7283a3fca1ab/dashboard/app/assets/javascripts/header.js#L454.  I'm not sure it's worth too much time refactoring since I'd like to rewrite both as React components.